src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,119 @@
+package duke;
+
+import java.util.Hashtable;
+import java.util.Scanner;
+import java.util.function.Consumer;
+
+public class Parser {
```

Comment:
Perhaps there should be a javadoc since this is a public class? I noticed the same issue in some of your other classes too.